### PR TITLE
Dev/generic type class

### PR DESCRIPTION
**Pull request by tao558**
This PR renames detected_object_type to class_map so it can be used more generically by other vital types. The vital type activity_type is also refactored out and replaced by this new generic class, as their implementations were very similar.

Lastly, a possible source of rep-exposure in kwiver::vital::activity::activity_type() was resolved

### CODE REVIEW

**Comment by tao558** [comment]
Jenkins test this please

File: vital/types/README.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 **camera_map** - Defines the interface base class camera_map and the subclass simple_camera_map, which allow representation of a group of camera instances.
 
+**class_map** -
```
Comment (as6520) [inline vital/types/README.md:9]: Minor nit: Could you add a description for `class_map` type?

Comment (tao558) [reply vital/types/README.md]: Done! Check out 0bfba0a7c388e9555081c4d3bf49228e4b966282. I also changed some of the wordage in the header, as it was specific to the detected_object type.

**Comment by tao558** [comment]
Jenkins test this please

**Comment by tao558** [comment]
Jenkins test this please

File: vital/types/class_map.h [hunk 2]
```diff
@@ -110,7 +110,7 @@ class VITAL_EXPORT detected_object_type
    * @param class_name Class name
    * @param score Probability score for the class
    */
-  detected_object_type( const std::string& class_name,
+  class_map( const std::string& class_name,
```
Comment (dstoup) [inline vital/types/class_map.h:113]: score seems to indented incorrectly now.

**Comment by mattdawkins** [comment]
Majorly an unrelated aside, though also related is: https://github.com/Kitware/kwiver/blob/master/vital/types/category_hierarchy.h I'm potentially interested in integrating with this class at some point in time

File: arrows/core/class_probablity_filter.cxx [hunk 3]
```diff
@@ -140,38 +140,38 @@ filter( const vital::detected_object_set_sptr input_set ) const
...
     bool det_selected( false );
-    auto out_dot = std::make_shared<vital::detected_object_type>( );
+    auto out_cm = std::make_shared<vital::class_map>( );
 
-    // Make sure that there is an associated DOT
-    auto input_dot = (*det)->type();
-    if ( ! input_dot )
+    // Make sure that there is an associated CM
+    auto input_cm = (*det)->type();
+    if ( ! input_cm )
```
[the remark this answers is not in the document]
Comment (tao558) [reply arrows/core/class_probablity_filter.cxx:147]: How does 02ef7ec9ae4fde8cf4f3fbe78eb62c28ecaeb4b9 look?

File: arrows/serialize/protobuf/tests/test_serialize.cxx [hunk 4]
```diff
@@ -224,7 +224,7 @@ TEST( serialize, detected_object_set )
   kwiver::vital::any obj_any( ser_dos_sptr );
   auto mes = obj_ser.serialize( obj_any );
 
-  // std::cout << "Serialized dot: \"" << *mes << "\"\n";
+  // std::cout << "Serialized cm: \"" << *mes << "\"\n";
```
[the remark this answers is not in the document]
Comment (tao558) [reply arrows/serialize/protobuf/tests/test_serialize.cxx]: Fixed in 4d1c1e786fb5b774dc9c8167660b299b6c5998af

File: vital/bindings/c/types/class_map.cxx [hunk 5]
```diff
@@ -30,107 +30,107 @@
...
+    "C::class_map::destroy", 0,
+    kwiver::vital_c::CM_SPTR_CACHE.erase( obj );
 
   );
 }
 
 
 // ------------------------------------------------------------------
-vital_detected_object_type_t* vital_detected_object_type_new_from_list( vital_detected_object_type_t* obj,
+vital_class_map_t* vital_class_map_new_from_list( vital_class_map_t* obj,
```
[the remark this answers is not in the document]
Comment (tao558) [reply vital/bindings/c/types/class_map.cxx:81]: Fixed in 38166d8088fe86ab562228517d379e694ecb473f

File: vital/types/class_map.h (hunk 2 above)
Comment (tao558) [reply vital/types/class_map.h:113]: Fixed in e45da5d1ba12fae92d21e0232281147b52860050. This happened in a different method declaration as well.

**Comment by tao558** [comment]
I'll add another commit to fix the copyrights, then assuming that those changes are approved, squash the post-review commits. I'll leave them separate for now to make it clearer what was changed when

File: arrows/core/class_probablity_filter.cxx [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*ckwg +29
- * Copyright 2016 by Kitware, Inc.
+ * Copyright 2016, 2020 by Kitware, Inc.
```
[the remark this answers is not in the document]
Comment (tao558) [reply arrows/core/class_probablity_filter.cxx]: Fixed in 6802e684083a1ec2ea5a2ac42995a140b35c4e42

**Comment by tao558** [comment]
Jenkins test this please

File: arrows/core/class_probablity_filter.cxx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dstoup) [reply arrows/core/class_probablity_filter.cxx:147]: Perfect, thanks!

**Pull request merged**